### ISSUE TITLE
Private `[Symbol.asyncIterator]` and `[Symbol.iterator]` methods incorrectly reported as unused

### ISSUE THREAD

**Issue by wojpawlik** [microsoft/TypeScript#42051]
**TypeScript Version:**  4.2.0-dev.20201219

**Search Terms:**

`Symbol.iterator` `Symbol.asyncIterator`

**Code**

```ts
export class Polling {
  private async *[Symbol.asyncIterator]() {}
  private *[Symbol.iterator]() {}

  async loop() {
    for await (const updates of this) {
      void updates
    }
  }
  syncLoop() {
    for (const updates of this) {
      void updates
    }
  }
}
```

**Expected behavior:**
No errors.

**Actual behavior:**

    '[Symbol.asyncIterator]' is declared but its value is never read.
    '[Symbol.iterator]' is declared but its value is never read.

[**Playground Link**](https://www.staging-typescript.org/play?noUnusedLocals=true&noUnusedParameters=true&target=6&ts=4.2.0-dev.20201219#code/KYDwDg9gTgLgBAYwDYEMDOa4AUJKQSwDsBzOAbwCg44wp8A3FGYOdAT0ITgCoBtAZTYBbAEa4AdO04BJZlCbQAugAoAlOQC+VGnUbMeA4WKTj8chVBXqyW7VK5IIEMGvLbqAM2isA7ijNwyggQhGjwAK5gACZMwJgQHnAwABb4aNbu1HD0EPhRcJExzGiZWtRlcGgcCAAyTi4ZWXBeUIHBoRHRsfGJKWmNTdm5+YXdpdpaWkA)
